src/calliope/backend/expression_parser.py

```diff
@@ -347,14 +346,20 @@ class EvalComparisonOp(EvalToArrayStr):
 
     OP_TRANSLATOR = {"<=": r" \leq ", ">=": r" \geq ", "==": " = "}
 
-    def __init__(self, tokens: pp.ParseResults) -> None:
+    def __init__(self, instring: str, loc: int, tokens: pp.ParseResults) -> None:
```

Comment:
It's a pyparsing logic anomoly. You can init a parsing class with one or three args and it will interpret them as `tokens` or [instring, loc, tokens] accordingly. `loc` is an int and we have ~100 comparisons opt classes created in a given model, so I expect the memory hit to negligble.